kernel/default.nix

```diff
@@ -0,0 +1,105 @@
+{
```

Comment:
The parent [Makefile](https://github.com/aws/aws-nitro-enclaves-sdk-bootstrap/blob/main/Makefile) should be modified to utilize Nix, as the current configuration will result in building the legacy 4.14 kernel version. Additionally, it is important to validate that this change does not adversely impact the Docker-based build process.

Reply:
Right now this change does not provide a way to build the new kernels with docker. I am not sure if we want to provide that or not. My view would be to completely go with one way and rework the repo to be completely nix based.

With this patch (similar to the linuxkit one) we add parallel infrastructure to build the new kernels. The old way is not touched by it for now.

My plan would be to have a top level setup that builds all the artifacts through nix and get rid of the docker based way. I intended to do that once I have all the pieces in place. What is missing at the moment is a nix-based build for `init`, which I have prepared, but need to test still.

What do you think?

Reply:
We can remove Docker from the build process. However, we can still utilize Nix within a Docker container, right?

In my opinion, the top-level build should be effortless. The number of CPU cores allocated for parallel builds should be determined automatically. If any dependencies need to be downloaded, the top-level build system should handle that seamlessly. Furthermore, if one wants to use this build in a containerized environment (e.g., CI/CD), employing Docker could still serve as a viable reference.

Reply:
All valid points, but in my mind they are to be addressed separately in later PRs.

I see this PR as a stepping stone to put a piece of the puzzle there for now, providing a way to build a more modern kernel in a reproducible way. Tying all the reproducible builds together into a simple way of building the "full set" of binaries would be the part of a separate PR. That separate PR would then also clean up the old way of building the binaries in my mind.

What is missing for this PR (and was so for the linuxkit one) is a reference to this on the top-level Readme. I will add some explanation there that this currently is in some transition phase where there is two distinct ways to build the binaries:
1) The traditional way where the build from the top level directory goes through the makefile and creates the binaries in a dockerized environment. 
2) The nix-based way that currently is lacking the integration into the whole package, but unblocks people that want to build more modern kernels right now, without waiting for the package to have done the transition completely.

Does that make sense or would you like to see this structured into one PR with multiple commits doing all of this in one go?